python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -325,7 +326,18 @@ def reset(self, errors_fatal=False):
         try:
             with open(self.config_path) as f:
                 new_config = yaml.safe_load(f.read())
-            validate_config(new_config)
+            if not new_config.get("no_config_validation_warning", False):
+                try:
+                    validate_config(new_config)
+                except Exception as e:
+                    if log_once("cluster_validate_config_fail"):
```

Comment:
Hmm actually I think we may want to log multiple times here. `reset` is called whenever the autoscaler config is updated, so I think it makes sense to re-warn if the new config is also invalid.

Reply:
The problem with logging multiple times is it warns every iteration of the monitor loop. It's a quite loud warning because it prints out the entire schema.

Reply:
ok, I added a check to only log the warning once per config change.

Reply:
ah shoot, i see what you mean. yeah printing this once per monitor loop will be very very annoying... how about an `if new_config != self.config`?

Reply:
already done!

Reply:
aaand this is a totally out of date snippet lol

thanks